src/main/java/org/datadog/jmxfetch/Instance.java

```diff
@@ -263,9 +263,26 @@ private void getMatchingAttributes() {
         LOGGER.info("Found " + matchingAttributes.size() + " matching attributes");
     }
 
-    private void refreshBeansList() throws IOException {
+    public LinkedList<String> getBeansScopes(){
+        if(this.beanScopes == null){
+            this.beanScopes = Configuration.getGreatestCommonScopes(configurationList);
+        }
+        return this.beanScopes;
+    }
 
-        this.beans = connection.queryMBeans();
+    private void refreshBeansList() throws IOException {
+        this.beans = new HashSet<ObjectInstance>();
+        try {
+            LinkedList<String> beanScopes = getBeansScopes();
+            for (String scope : beanScopes) {
+                ObjectName name = new ObjectName(scope);
+                this.beans.addAll(connection.queryMBeans(name));
+            }
+        }
+        catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Good idea. 

Otherwise I can't think of a specific exception that i want to catch. I added this very generic `try {} catch(){}` statement in order to fallback in the previous behavior in case anything was turning wrong. I should maybe remove it completely.